bin/propolis-standalone/src/main.rs

```diff
@@ -952,6 +952,42 @@ fn generate_smbios(params: SmbiosParams) -> anyhow::Result<smbios::TableBytes> {
     Ok(smb_tables.commit())
 }
 
+fn generate_bootorder(config: &config::Config) -> anyhow::Result<fwcfg::Entry> {
+    let names = config.main.boot_order.as_ref().unwrap();
+
+    let mut order = fwcfg::formats::BootOrder::new();
+    for name in names.iter() {
+        let dev = config
+            .devices
+            .get(name)
+            .ok_or(anyhow::anyhow!("Could not find device: {name}"))?;
+
+        let get_pci_path = || {
+            dev.options
+                .get("pci-path")
+                .and_then(|v| v.as_str())
+                .and_then(config::parse_bdf)
+                .expect("PCI device has valid BDF")
+        };
+
+        match dev.driver.as_str() {
+            "pci-virtio-block" => {
+                order.add_disk(get_pci_path().location);
+            }
+            "pci-nvme" => {
+                order.add_nvme(get_pci_path().location, 0);
+            }
+            driver if driver.starts_with("pci-") => {
+                order.add_pci(get_pci_path().location, "device");
+            }
+            dev => {
+                anyhow::bail!("Unsupported boot device type: {dev}");
```

Comment:
it might be worth including the name of the device that was a type we didn't support? maybe something like:
```suggestion
                anyhow::bail!("Boot device '{name}' is an unsupported device type: {dev}");
```